test/Microsoft.ApplicationInsights.AspNet.Tests/Helpers/HttpContextAccessorHelper.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Mvc;
+using Microsoft.Framework.DependencyInjection;
+using System;
+
+namespace Microsoft.ApplicationInsights.AspNet.Tests.Helpers
+{
+    public static class HttpContextAccessorHelper
+    {
+        public static HttpContextAccessor CreateHttpContextAccessor(RequestTelemetry requestTelemetry = null, ActionContext actionContext = null)
```

Comment:
I like how this new method makes tests shorter. On the other hand, it's also a good example of why *Service Locator* is an anti-pattern. 
- This logic is quite complex for something as simple as a telemetry initializer test.
- The `RequestTelemetry` and `ActionContext` dependencies are hidden in the `RequestServices` service provider.
- The tests would be much simpler and clearer if the components under tests used constructor injection instead.